workers/unity/Packages/com.improbable.gdk.gameobjectcreation/WorkerEntityGameObjectLinkerSystem.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.ObjectModel;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.GameObjectRepresentation;
+using Improbable.Worker;
+using Improbable.Worker.Core;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.GameObjectCreation
+{
+    [DisableAutoCreation]
+    [UpdateInGroup(typeof(SpatialOSReceiveGroup.GameObjectInitializationGroup))]
+    public class WorkerEntityGameObjectLinkerSystem : ComponentSystem
+    {
+        private struct WorkerConnectedData
+        {
+            public readonly int Length;
+            [ReadOnly] public ComponentDataArray<WorkerEntityTag> DenotesWorkerEntity;
+            [ReadOnly] public ComponentDataArray<OnConnected> DenotesJustConnected;
+            public EntityArray Entities;
+        }
+
+        private struct WorkerDisconnectedData
+        {
+            public readonly int Length;
+            [ReadOnly] public ComponentDataArray<WorkerEntityTag> DenotesWorkerEntity;
+            [ReadOnly] public SharedComponentDataArray<OnDisconnected> DenotesJustDisconnected;
+            public EntityArray Entities;
+        }
+
+        [Inject] private WorkerConnectedData connectedData;
+        [Inject] private WorkerDisconnectedData disconnectedData;
+
+        [Inject] private EntityGameObjectLinkerSystem linkerSystem;
+        [Inject] private WorkerSystem workerSystem;
+
+        private ViewCommandBuffer viewCommandBuffer;
+        private GameObject gameObjectToLink;
+
+        public WorkerEntityGameObjectLinkerSystem(GameObject workerGameObject)
+        {
+            gameObjectToLink = workerGameObject;
+        }
+
+        protected override void OnCreateManager()
+        {
+            viewCommandBuffer = new ViewCommandBuffer(EntityManager, workerSystem.LogDispatcher);
+        }
+
+        protected override void OnUpdate()
+        {
+            for (int i = 0; i < connectedData.Length; i++)
+            {
+                linkerSystem.Linker.LinkGameObjectToEntity(gameObjectToLink, connectedData.Entities[i],
+                    new EntityId(0), viewCommandBuffer);
+            }
+
+            for (int i = 0; i < disconnectedData.Length; i++)
+            {
+                linkerSystem.Linker.UnlinkGameObjectFromEntity(gameObjectToLink, disconnectedData.Entities[i],
```

Comment:
Note for docs - we don't delete the gameobject. I think that is fine. But it should be noted and probably added to the connector stuff.

Reply:
Yes I was considering which way is better and decided to go with this one in the end, since it's created by the user in the first place.

Reply:
For the connectors, I didn't add it yet because I couldn't think of anything sensible to add to the playground and I'm confident it works because of the unit tests. Should I just add a blank GameObject to the workers in Playground, or should I try to come up with something good to add there?

Reply:
I think what you have here is fine. We should think about adding something in a follow up though. We could rewrite a player lifecycle in it if we wanted to.